### PR TITLE
Feature/credit delegation

### PR DESCRIPTION
**Pull request by majkic99**
getCreditDelegation view helper function currently in action itself, but can be transfered to view if it makes more sense

### CODE REVIEW

File: contracts/actions/spark/SparkDelegateCredit.sol [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes calldata _callData,
+        bytes32[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+
+        params.market = _parseParamAddr(params.market, _paramMapping[0], _subData, _returnValues);
```
Comment (yippee-ki-yay) [inline contracts/actions/spark/SparkDelegateCredit.sol:34]: wat

File: contracts/actions/aaveV3/AaveV3DelegateCredit.sol [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    function executeActionDirect(bytes calldata _callData) public payable override {
+        Params memory params = parseInputs(_callData);
+        (bytes memory logData) = _delegate(params);
+        logger.logActionDirectEvent("AaveV3DelegateCredit", logData);
+    }
+
+    function executeActionDirectL2() public payable {
+        Params memory params = decodeInputs(msg.data[4:]);
+        (bytes memory logData) = _delegate(params);
+        logger.logActionDirectEvent("SparkDelegateCredit", logData);
```
Comment (yippee-ki-yay) [inline contracts/actions/aaveV3/AaveV3DelegateCredit.sol:58]: AaveV3

**Review by yippee-ki-yay — COMMENTED** [review]
Few small fixes needed

**Review by kjaredic — CHANGES_REQUESTED** [review]
lgtm just one small change

File: test/aaveV3/aave-delegate-credit.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const { redeploy } = require('../utils');
+const { aaveV3DelegateCreditTest } = require('./aave-tests');
+
+describe('Aave-Delegate-Credit', function () {
+    this.timeout(150000);
+
+    before(async () => {
+        await redeploy('AaveV3DelegateCredit');
+    });
+    it('... should run full aave set EMode test', async () => {
```
Comment (yippee-ki-yay) [inline test/aaveV3/aave-delegate-credit.js:10]: Wrong description

**Review by yippee-ki-yay — APPROVED** [review]

**Review by yippee-ki-yay — APPROVED** [review]

**Pull request merged**